File: C++/프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer(2);
    int area = brown + yellow;
    int j;
    for(int i = area; i >= 0; i--) {
        if(area % i != 0) continue;
        j = area / i;
        if((i - 2) * (j - 2) == yellow){
            answer[0] = i;
            answer[1] = j;
            break;
        }
    }
    return answer;
}

File: Python3/프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    area = brown + yellow
    c = 0
    for r in range(area, 0, -1):
        if area % r != 0: continue
        c = area / r
        if (c - 2) * (r - 2) == yellow:
            return [r, c]
            
    
    return [-1, -1]